controllers/templatesync/template_sync.go

```diff
@@ -335,28 +391,49 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Requ
 
 		dependencyFailures := r.processDependencies(ctx, dClient, rMapper, templateDeps, tLogger)
 
-		// fetch resource
-		res := dClient.Resource(rsrc).Namespace(instance.GetNamespace())
+		// Instantiate a dynamic client -- if it's a clusterwide resource, then leave off the namespace
+		var res dynamic.ResourceInterface
+
+		// Now that there is a mapping, there is a definitive answer whether it's cluster scoped rather
+		// than the previous educated guess based on the provided group and kind.
+		isClusterScoped = mapping.Scope == meta.RESTScopeRoot
+		resourceNs := ""
+
+		if !isClusterScoped {
+			resourceNs = instance.GetNamespace()
+		}
+
+		res = dClient.Resource(rsrc).Namespace(resourceNs)
```

Comment:
Woah, `.Namespace("")` works? I feel like we have something similar to what this is doing here in another controller... but we didn't do it this way. I don't know if either approach is better than the other...

Reply:
Config-policy-controller does this:
```
var res dynamic.ResourceInterface
if namespaced {
	res = r.TargetK8sDynamicClient.Resource(mapping.Resource).Namespace(object.Object.Metadata.Namespace)
} else {
	res = r.TargetK8sDynamicClient.Resource(mapping.Resource)
}
```

Reply:
Yeah, I thought it wasn't working until I tried it again. I don't think I have a big preference--they're pretty similar.